modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/HeaderInformation.java

```diff
@@ -0,0 +1,823 @@
+ *
+ *         request("request-address", "Hello with Info", requestInfos).onSuccess(message -> {
+ *             var responseInfos = HeaderInformation.from(message);
+ *
+ *             logger.info("Received version: {}", responseInfos.getInt("version", -1));
+ *             logger.info("Received log enabled state: {}", responseInfos.getBoolean("log-enabled", false));
+ *             logger.info("Received initial char: {}", responseInfos.getChar("my-initial-char", '\0'));
+ *             logger.info("Received coworker name: {}", responseInfos.get("my-coworkers-name", "unset"));
+ *         });
+ *     }
+ * }
+ * }
+ * </pre>
+ *
+ * @author Cedric Boes (@cb0s), Ludwig Richter (@fussel178)
+ */
+public class HeaderInformation {
+
+	/**
+	 * Merges an array of header information into one header information object.
+	 *
+	 * @param information multiple header information that should be merged into one
+	 * @return the merged header information
+	 */
+	public static HeaderInformation merge(HeaderInformation... information) {
+		return merge(Arrays.stream(information));
+	}
+
+	/**
+	 * Merges a list of header information into one header information object.
+	 *
+	 * @param list multiple header information that should be merged into one
+	 * @return the merged header information
+	 */
+	public static HeaderInformation merge(List<HeaderInformation> list) {
+		return merge(list.stream());
+	}
+
+	/**
+	 * Merges a stream of header information into one header information object.
+	 *
+	 * @param stream a stream that contains multiple header information that should be merged into one
+	 * @return the merged header information
+	 */
+	public static HeaderInformation merge(Stream<HeaderInformation> stream) {
+		var headers = stream.map(HeaderInformation::getHeaders);
+		return new HeaderInformation(MultiMapUtils.merge(headers));
+	}
+
+	/**
+	 * Creates new header information from existing {@link MultiMap Vert.x headers}.
+	 *
+	 * @param headers the basic {@link MultiMap Vert.x headers} that should be wrapped into the header information
+	 * @return new header information with the wrapped {@link MultiMap Vert.x headers}
+	 */
+	public static HeaderInformation from(MultiMap headers) {
+		return new HeaderInformation(headers);
+	}
+
+	/**
+	 * Creates new header information from existing {@link Message Vert.x message}.
+	 *
+	 * @param message the received {@link Message Vert.x message} which contains the received headers
+	 * @return new header information with the received headers from the {@link Message Vert.x message}
+	 */
+	public static HeaderInformation from(Message<?> message) {
+		return new HeaderInformation(message);
+	}
+
+	/**
+	 * Creates new header information from existing {@link DeliveryOptions}.
+	 *
+	 * @param options existing {@link DeliveryOptions} which contain the basic {@link MultiMap Vert.x headers}
+	 * @return new header information with the headers from the {@link DeliveryOptions}
```

Comment:
```suggestion
	 * Creates a new header information object from existing {@link DeliveryOptions}.
	 *
	 * @param options {@link DeliveryOptions} that contain the {@link MultiMap Vert.x headers}
	 * @return header information object with the headers from the {@link DeliveryOptions}
```

Reply:
Done in [`2cd29d6` (#484)](https://github.com/wuespace/telestion-core/pull/484/commits/2cd29d6f1426056b991dcbdcda1b3d2c3274fe9e)